src/Sentry.Unity.iOS/SentryCocoaBridgeProxy.cs

```diff
@@ -11,10 +13,74 @@ namespace Sentry.Unity.iOS
     /// <see href="https://github.com/getsentry/sentry-cocoa"/>
     internal static class SentryCocoaBridgeProxy
     {
-        [DllImport("__Internal")]
+        public static bool Init(SentryUnityOptions options)
+        {
+            // Note: used on macOS only
+            if (LoadLibrary() != 1)
+            {
+                return false;
+            }
+
+            var cOptions = OptionsNew();
+
+            // Note: DSN is not null because options.IsValid() must have returned true for this to be called.
+            OptionsSetString(cOptions, "dsn", options.Dsn!);
+
+            if (options.Release is not null)
+            {
+                options.DiagnosticLogger?.LogDebug("Setting Release: {0}", options.Release);
+                OptionsSetString(cOptions, "release", options.Release);
+            }
+
+            if (options.Environment is not null)
+            {
+                options.DiagnosticLogger?.LogDebug("Setting Environment: {0}", options.Environment);
+                OptionsSetString(cOptions, "environment", options.Environment);
+            }
+
+            options.DiagnosticLogger?.LogDebug("Setting Debug: {0}", options.Debug);
+            OptionsSetInt(cOptions, "debug", options.Debug ? 1 : 0);
+
+            var diagnosticLevel = options.DiagnosticLevel.ToString().ToLowerInvariant();
+            options.DiagnosticLogger?.LogDebug("Setting DiagnosticLevel: {0}", diagnosticLevel);
+            OptionsSetString(cOptions, "diagnosticLevel", diagnosticLevel);
+
+            // Disabling the native in favor of the C# layer for now
+            options.DiagnosticLogger?.LogDebug("Disabling native auto session tracking");
+            OptionsSetInt(cOptions, "enableAutoSessionTracking", 0);
+
+            options.DiagnosticLogger?.LogDebug("Setting SendDefaultPii: {0}", options.SendDefaultPii);
+            OptionsSetInt(cOptions, "sendDefaultPii", options.SendDefaultPii ? 1 : 0);
+
+            options.DiagnosticLogger?.LogDebug("Setting MaxBreadcrumbs: {0}", options.MaxBreadcrumbs);
+            OptionsSetInt(cOptions, "maxBreadcrumbs", options.MaxBreadcrumbs);
+
+            options.DiagnosticLogger?.LogDebug("Setting MaxCacheItems: {0}", options.MaxCacheItems);
+            OptionsSetInt(cOptions, "maxCacheItems", options.MaxCacheItems);
+
+            StartWithOptions(cOptions);
```

Comment:
`m`: I think we could add a method wrapping a call to `CFRelease` for the `cOptions`, as we don't need the dictionary anymore after starting the SDK.

Reply:
the ownership is returned by `__bridge_transfer` here:

```objective-c
void SentryNativeBridgeStartWithOptions(const void *options)
{
    NSMutableDictionary *dictOptions = (__bridge_transfer NSMutableDictionary *)options;
    [SentrySDK performSelector:@selector(startWithOptions:) withObject:dictOptions];
}
```

and AFAIU the object will be freed after startWithOptions: is done with it. Therefore we can't do it again manually with CFRelease() or it would be a double-free. But that's just my limited understanding of memory management in objective-c...

Reply:
Thanks for the detailed explanation. Never had to do manual ARC yet.